thunder/core/proxies.py

```diff
@@ -1880,6 +1880,111 @@ def real(self):
         return method(self)
 
 
+class SubclassTensorProxy(TensorProxy):
```

Comment:
this should be able to express `__tensor_flatten__(self) -> tuple[list[str], dict[str, Any]]` (and `__tensor_unflatten__(inner_tensors: dict[str, Tensor], metadata: dict[str, Any], outer_size, outer_stride) -> MySubclass`).
For it to happen, somewhere I have to give instances of this class the attribute names of tensors and non-tensor values.

Reply:
Would you add a comment describing what this class is intended for?